examples/poiseuille-mpi.py

```diff
@@ -164,26 +166,39 @@ def poiseuille_soln(nodes, eos, cv=None, **kwargs):
     if rank == 0:
         logger.info(init_message)
 
-    get_timestep = partial(inviscid_sim_timestep, discr=discr, t=current_t,
-                           dt=current_dt, cfl=current_cfl, eos=eos,
-                           t_final=t_final, constant_cfl=constant_cfl)
-
     def my_rhs(t, state):
         return ns_operator(discr, eos=eos, boundaries=boundaries, cv=state, t=t)
 
     def my_checkpoint(step, t, dt, state):
+        t_remaining = max(0, t_final - t)
+        checkpoint_cfl = current_cfl
+        viz_fields = []
+
+        if constant_cfl is True:
+            dt = (
+                current_cfl * get_viscous_timestep(discr=discr, eos=eos, cv=state)
+            )
+            from grudge.op import nodal_min
+            dt = nodal_min(discr, "vol", dt)
+        else:
+            from mirgecom.viscous import get_viscous_cfl
+            cfl_field = get_viscous_cfl(discr, eos, dt, state)
+            viz_fields.append(("cfl", cfl_field))
+            from grudge.op import nodal_max
+            checkpoint_cfl = nodal_max(discr, "vol", cfl_field)
+
+        dt = min(dt, t_remaining)
         return sim_checkpoint(discr, visualizer, eos, cv=state,
-                              vizname=casename, step=step,
+                              vizname=casename, step=step, cfl=checkpoint_cfl,
```

Comment:
`checkpoint_cfl` needs to get defined even for constant CFL mode. Something like...
`checkpoint_cfl = current_cfl` somewhere near the top of the routine or just before the block where CFL (or DT) is calculated.

Reply:
AFAIK, it is. It's first defined on line 174 as current CFL, which should be correct for constant CFL mode.